DatadogInternal/Sources/Context/AppState.swift

```diff
@@ -142,31 +151,66 @@ extension AppStateHistory {
     }
 }
 
-#if canImport(UIKit)
-
-import UIKit
-
 #if canImport(WatchKit)
+
 import WatchKit
 
-public typealias ApplicationState = WKApplicationState
-#else
-public typealias ApplicationState = UIApplication.State
-#endif
+public struct DefaultAppStateProvider: AppStateProvider {
+    public init() {}
+
+    /// Gets the current application state.
+    ///
+    /// **Note**: Must be called on the main thread.
+    public var current: AppState {
+        let wkState = WKExtension.dd.shared.applicationState
+        return AppState(wkState)
+    }
+}
 
 extension AppState {
-    public init(_ state: ApplicationState) {
+    public init(_ state: WKApplicationState) {
         switch state {
-        case .active:
-            self = .active
-        case .inactive:
-            self = .inactive
-        case .background:
-            self = .background
+        case .active: self = .active
+        case .inactive: self = .inactive
+        case .background: self = .background
         @unknown default:
-            self = .active // in case a new state is introduced, we rather want to fallback to most expected state
+            self = .active // in case a new state is introduced, default to most expected state
         }
     }
 }
 
+#elseif canImport(UIKit)
+
+import UIKit
+
+public struct DefaultAppStateProvider: AppStateProvider {
+    public init() {}
+
+    /// Gets the current application state.
+    ///
+    /// **Note**: Must be called on the main thread.
+    public var current: AppState {
+        let uiKitState = UIApplication.dd.managedShared?.applicationState ?? .active // fallback to most expected state
+        return AppState(uiKitState)
+    }
+}
+
+extension AppState {
+    public init(_ state: UIApplication.State) {
+        switch state {
+        case .active: self = .active
+        case .inactive: self = .inactive
+        case .background: self = .background
+        @unknown default: self = .active // in case a new state is introduced, default to most expected state
+        }
+    }
+}
+
+#else // macOS (no UIKit and no WatchKit)
+
+public struct DefaultAppStateProvider: AppStateProvider {
+    public init() {}
+    public let current: AppState = .active
```

Comment:
I'm wondering if it might better to use `.background` as the fallback state of `.active`. This could potentially avoid false positives, which might contributed to issues like the 1ns sessions. What happens in the case of cold starts and pre-warmings? Is the state correctly initialized?

Reply:
Please mind that this default is only for macOS where we don't offer official support. Because we don't really test SDK for macOS and knowing that we do not subscribe to `AppKit's` lifecycle notifications anywhere in the SDK, always assuming `.active` may seem safer. In macOS apps, this state will very likely become the only one that is assumed (which is fine for non-official support and no RUM offering for macOS).